sentry/src/main/java/io/sentry/event/interfaces/SentryStackTraceElement.java

```diff
@@ -64,17 +74,56 @@ public String getPlatform() {
         return platform;
     }
 
+    public Map<String, Object> getLocals() {
+        return locals;
+    }
+
     /**
      * Convert an array of {@link StackTraceElement}s to {@link SentryStackTraceElement}s.
      *
      * @param stackTraceElements Array of {@link StackTraceElement}s to convert.
      * @return Array of {@link SentryStackTraceElement}s.
      */
     public static SentryStackTraceElement[] fromStackTraceElements(StackTraceElement[] stackTraceElements) {
+        Frame[] localsCache = LocalsCache.getCache();
+
+        /*
+        Verify the localsCache stacktrace length and method classes/names match the stackTraceElements
+        length and method classes/names. This needs to be done in its own loop because the entire stack
+        must be match before we can assume the stored locals are most likely from the stacktrace that
+        is being recorded.
+
+        The reason they might *not* match is that a user can stash an exception in a variable, allow
+        another exception to occur, and then attempt to send the first exception to Sentry. Since we
+        can't cache every exception's locals for all time, we attempt a best-effort where we store the
+        last invocation and do this matching to see if they're most likely from the same call.
+
+        This code is only run if:
+        1. the JVM is running our agent (and therefore LocalsCache is not null)
+        2. the length of the cached locals matches the length of the stacktrace that is being sent
+         */
```

Comment:
As discussed in Slack, yeah, chained exceptions will be an issue in the current impl.